src/app/games/yinsh/Yinsh.component.ts

```diff
@@ -0,0 +1,468 @@
+import { Coord } from 'src/app/jscaip/Coord';
+import { MoveEncoder } from 'src/app/jscaip/Encoder';
+import { HexaDirection } from 'src/app/jscaip/HexaDirection';
+import { HexaLayout } from 'src/app/jscaip/HexaLayout';
+import { FlatHexaOrientation } from 'src/app/jscaip/HexaOrientation';
+import { Player } from 'src/app/jscaip/Player';
+import { MessageDisplayer } from 'src/app/services/message-displayer/MessageDisplayer';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPValidation } from 'src/app/utils/MGPValidation';
+import { YinshFailure } from './YinshFailure';
+import { YinshGameState } from './YinshGameState';
+import { YinshLegalityStatus } from './YinshLegalityStatus';
+import { YinshMinimax } from './YinshMinimax';
+import { YinshCapture, YinshMove } from './YinshMove';
+import { YinshPiece } from './YinshPiece';
+import { YinshRules } from './YinshRules';
+
+interface CaseInfo {
+    coord: Coord,
+    coordinates: string,
+    caseClasses: string[],
+    markerClasses: string[],
+    ringClasses: string[],
+    isMarker: boolean,
+    isRing: boolean,
+    center: Coord,
+    removedClass: string,
+}
+
+interface ViewInfo {
+    caseInfo: CaseInfo[][],
+    selectableCoords: Coord[],
+    selectedCoords: Coord[],
+    targets: Coord[],
+    markerSize: number,
+    indicatorSize: number,
+    ringOuterSize: number,
+    ringMidSize: number,
+    ringInnerSize: number,
+    sideRings: [number, number],
+    sideRingClass: [string, string],
+}
+
+@Component({
+    selector: 'app-yinsh',
+    templateUrl: './yinsh.component.html',
+    styleUrls: ['../../components/game-components/abstract-game-component/abstract-game-component.css'],
+})
+export class YinshComponent extends HexagonalGameComponent<YinshMove, YinshGameState, YinshLegalityStatus> {
+    private static RING_OUTER_SIZE: number = 40;
+    private static RING_MID_SIZE: number = 34;
+    private static RING_INNER_SIZE: number = 28;
+    private static MARKER_SIZE: number = YinshComponent.RING_INNER_SIZE;
+    private static INDICATOR_SIZE: number = 10;
+
+    public rules: YinshRules = new YinshRules(YinshGameState);
+
+    public hexaLayout: HexaLayout = new HexaLayout(YinshComponent.RING_OUTER_SIZE * 1.50,
+                                                   new Coord(YinshComponent.RING_OUTER_SIZE * 2, 0),
+                                                   FlatHexaOrientation.INSTANCE);
+
+    public encoder: MoveEncoder<YinshMove> = YinshMove.encoder;
+
+    public scores: number[] = [0, 0];
+
+    private constructedState: YinshGameState;
+
+    private movePhase: 'INITIAL_CAPTURE_SELECT_FIRST' | 'INITIAL_CAPTURE_SELECT_LAST' | 'INITIAL_CAPTURE_SELECT_RING' | 'MOVE_START' | 'MOVE_END' | 'FINAL_CAPTURE_SELECT_FIRST' | 'FINAL_CAPTURE_SELECT_LAST' | 'FINAL_CAPTURE_SELECT_RING'
+        = 'MOVE_START';
+
+    private moveStart: MGPOptional<Coord> = MGPOptional.empty();
```

Comment:
up to date oO ?
On avait pas déjà modifié ça ?

Reply:
Yep, repulled develop